docs/csharp/language-reference/compiler-messages/cs8352.md

```diff
@@ -0,0 +1,120 @@
+---
+description: "Compiler Error CS8352"
+title: "Compiler Error CS8352"
+ms.date: 12/05/2023
+f1_keywords:
+  - "CS8352"
+helpviewer_keywords:
+  - "CS8352"
+---
+# Compiler Error CS8352
+
+Cannot use variable 'variable' in this context because it may expose referenced variables outside of their declaration scope.
+
+This error indicates potential undefined behavior when there's a risk of accessing a stack allocated variable after it has already been deleted from the stack.
+
+## To correct this error
+
+Ways to avoid `CS8352` are, for example:
+
+* Redesign the architecture of your solution regarding the way objects are allocated in the memory and the memory management in your implementation.
+* Use `in` keyword and create the object in wider scope
+* Use `new` keyword to create your object on a heap instead of a stack
+* Use `ref` and `unsafe` keyword to notify the compiler of intentional object passing
+
+## Example
+
+The following sample generates CS8352:
+
+```csharp
+class Program
+{
+    static public Span<int> CreateSpanWithValue(int size, int value)
+    {
+        Span<int> localSpan = stackalloc int[size];
+        localSpan[0] = value;
+        return localSpan;    // CS8352
+    }                        // localSpan object is deleted at this point
+
+    static public void Main()
+    {
+        Span<int> localSpan = CreateSpanWithValue(5, 10);
+        localSpan[2] = 14;
+        foreach(var item in localSpan)
+        {
+            Console.WriteLine(item.ToString());  // Reading from deleted object
+        }
+    }
+    // Example of output:
+    // 284945320
+    // 149
+    // 149
+    // 1369466325
+    // 284945320
+}
+```
+
+The following sample uses `in` keyword and creates the object in wider scope:
+
+```csharp
+class Program
+{
+    static public Span<int> FillSpanWithValue(in Span<int> span, int value)
+    {
+        span[0] = value;
+    }
+
+    static public void Main()
+    {
+        Span<int> localSpan = stackalloc int[5];
+        FillSpanWithValue(in localSpan, 10);
+        localSpan[2] = 14;
+        foreach(var item in localSpan)
+        {
+            Console.WriteLine(item.ToString());
+        }
+    }
+    // Output:
+    // 10
+    // 0
+    // 14
+    // 0
+    // 0
+}
+```
```

Comment:
The `in` modifier has nothing to do with the warning / error.  See this [sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAAwCwAUEQIwVEAE+VA7BQN4V2cNUBsDAzAwAsdAGJQANhIDKABwgwA6lAQALAGoQJqAKYAKOQoA8sBAD46AZ3nw6pugDctugJQcu7cl29WbAbWIAXToAXkdnHQBud04AXwoY7j58QXwRAFloGD03Lw9E70MYExhzOgkAewBjLSLQqwQIKoBrLUqqu1K/AFZA6LyfMUkZG2U1TW19dtqbODoqYhd+wc5pkYU/QmCwqiFllYAzCuQdJtU9J2Q7BB0wTvLqmYVclbpPV+9GAE49FVuAOgAKhVpAhkLAAOY5JYFLjxAZ0eHeAD0yLoAHlUAhZFiQIlUfMyAiCUSUWjdvi0aSuCSKLEgA===) which is the same sample you have, but without `in`.

That's because (see earlier comment) the problem isn't that the `Span` has been reclaimed. Instead, it's because the memory referred to in the `Span` object has been reclaimed. (Both examples work because the Span refers to memory allocated in the calling context, not in the called method.)